util/util.go

```diff
@@ -191,41 +200,62 @@ func (s server) GetGRPCServer(opts ...grpc.ServerOption) *grpc.Server {
 	return grpc.NewServer(opts...)
 }
 
-func readContent(path string) error {
-	cont, err := os.Open(path)
+func InitilizegRPCServer(conf *model.Config) (*server, error) {
+
+	store, err := database.NewStore(conf)
+
 	if err != nil {
-		log.Fatal(err)
-		return err
+		return nil, err
 	}
-	_, err = io.Copy(os.Stdout, cont)
-	if err != nil {
-		log.Fatal(err)
-		return err
+
+	s := &server{
+		store:   store,
+		auth:    NewAuthenticator(conf.SigninKey, conf.AuthKey),
+		tls:     conf.TLS.Enabled,
 	}
-	return nil
+	return s, nil
 }
 
+func NewConfigFromFile(path string) (*model.Config, error) {
+	f, err := ioutil.ReadFile(path)
+	if err != nil {
+		return nil, err
+	}
+
+	var c model.Config
+	err = yaml.Unmarshal(f, &c)
+	if err != nil {
+		return nil, err
+	}
 
+	if c.Host == "" {
+		log.Println("Host not provided in the configuration file")
+		c.Host = "localhost:50051"
+	}
 
-func InitilizegRPCServer() *server {
+	if c.SigninKey == "" {
+		log.Println("SigninKey not provided in the configuration file")
+		c.Host = "dev-env"
+	}
 
-	store, err := database.NewStore()
+	if c.AuthKey == "" {
+		log.Println("AuthKey not provided in the configuration file")
+		c.Host = "development-environment"
+	}
 
-	if err != nil {
-		log.Fatalf("failed to connect to database: %v", err)
+	if c.DB.Host == "" || c.DB.User == "" || c.DB.Pass == "" || c.DB.Name == "" {
+		return nil, errors.New("DB paramenters missing in the configuration file")
 	}
-	// todo: change handling of tls, the example below not good enough
 
-	tls := true
-	mode := os.Getenv("SSL_OFF")
-	if strings.ToLower(mode) == "true" {
-		tls = false
+	if c.DB.Port == 0 {
+		c.DB.Port = 5432
 	}
 
-	s := &server{
-		store:   store,
-		auth:    NewAuthenticator(os.Getenv("SIGNIN_KEY")),
-		tls:     tls,
+	if c.TLS.Enabled {
+		if c.TLS.CAFile == "" || c.TLS.CertKey == "" || c.TLS.CertFile == "" {
+			return nil, errors.New("Provide Certificates in the config file")
+		}
```

Comment:
Same, as default value we can take into certificates which are exists under `tests`, so if those certificates provided as default, we do not need to give back error.